### PR TITLE
Support Monorepo's

### PR DESCRIPTION
**Pull request by dawidvdh**
I currently have a react native project within a monorepo that also contains the api and web app.

Therefore fetching the git history specific to the react native directory is required for this plugin to work effectively. I don't imagine any issues from this change however @xotahal would love to get your input on the approach and if you perhaps have a better idea on how to achieve this.

### CODE REVIEW

**Comment by xotahal** [comment]
I was thinking about this as well. But wouldn't be better to use scope from commits?

```
fix(mobile): ...
fix(web): ...
fix(mobile): ..
```

Then if you say I want only mobile scope it would skip the web commit in this case.

**Comment by dawidvdh** [comment]
Ah no way, I had exactly the same thought but what happens when you need to scope a fix or feat platform specifically for example `ios(fix): ...` or `android(feat): ...`?

**Comment by xotahal** [comment]
Yeah, so this would be a parameter of the script. So when you build ios you would set `mobile` and `ios` as scopes you want to "include" in this build. When you building android you would set `mobile` and `android`. I think it makes sense ...

**Comment by dawidvdh** [comment]
@xotahal okay I have added a paramater of `include_scopes` which will only allow the scopes that are added, let me know if this is what you had in mind, the tests are passing however there is a rubocop complaint about a function being too complicated.

**Review by xotahal — APPROVED** [review]

File: lib/fastlane/plugin/semantic_release/actions/analyze_commits.rb [hunk 1]
```diff
@@ -315,6 +315,20 @@ def self.available_options
             description: "To parse version number from tag name",
             default_value: '\d+\.\d+\.\d+'
           ),
+          FastlaneCore::ConfigItem.new(
+            key: :prevent_tag_fallback,
```
Comment (xotahal) [inline lib/fastlane/plugin/semantic_release/actions/analyze_commits.rb:319]: How is this related to the feature?

Comment (dawidvdh) [reply lib/fastlane/plugin/semantic_release/actions/analyze_commits.rb:319]: I think the ability to opt out of falling back to `vX.X.X` as the tag is crucial for monorepos, especially if tags like v1.0.2 was created on the repository for a certain package that has nothing to do with the mobile-app.

I have a repository that I have inherited with tags that already exist like `v1.0.4` and now I am creating the mobile app package and do not want it to start from that tag that has nothing to do with mobile, therefore I would like to opt out.

**Comment by dawidvdh** [comment]
Hey @xotahal will this be merged?
